dcos_test_utils/upgrade.py

```diff
@@ -125,17 +127,26 @@ def upgrade_dcos(
                     '--speed-limit', '100000',
                     '--speed-time', '60',
                     '--remote-name', upgrade_script_path])
-            ssh_client.command(host, ['sudo', 'bash', 'dcos_node_upgrade.sh'])
-            wait_metric = {
-                'master': 'registrar/log/recovered',
-                'slave': 'slave/registered',
-                'slave_public': 'slave/registered',
-            }[role]
-            log.info('Waiting for {} to rejoin the cluster...'.format(role_name))
-            try:
-                wait_for_mesos_metric(dcos_api_session, host, wait_metric)
-            except retrying.RetryError as exc:
-                raise Exception(
-                    'Timed out waiting for {} to rejoin the cluster after upgrade: {}'.
-                    format(role_name, repr(host))
-                ) from exc
+            if use_checks == 'true':
+                try:
+                    output = ssh_client.command(host, ['sudo', 'bash', 'dcos_node_upgrade.sh'])
+                    log.info("Output from checks" + output.decode("utf-8"))
+                except CalledProcessError as e:
```

Comment:
There is no reason to catch a CalledProcessed error in this manner. Also, it is rarely advisable to raise a new exception from a different exception without using `raise from`. If your goal is to make sure you have access to this output, I would suggest the following:
```
log.info("Starting upgrade script on {host} ({role_name})...".format(host=host, role_name=role_name))
if use_checks:
    ssh_client.command(host, ['sudo', 'bash', 'dcos_node_upgrade.sh'], stdout=sys.stdout.buffer)
else:
    ssh_client. ... [keep it the same hereafter]
```

This is much more robust as you are going to get the stdout on the driver machine instantly whereas the current implementation only works when the process actually finishes (which is far from guaranteed in testing)